### ISSUE TITLE
Bake To Keyframes Stops at frame 250

### ISSUE THREAD

**Issue by FlynnD273** [JacquesLucke/animation_nodes#1507]
I'm unsure whether this is a problem for all blend files, but I made a spectrum visualizer from an audio file, with 128 cubes being copied then moved up based on the amplitude of the frequency from the spectrum node. When I press the "bake to keyframes" button, it'll bake just fine, then stop at frame 250. My animation is 7195 frames long. The workaround I have is to manually set the execution mode to "bake," then run the animation all the way until the end. So there is a workaround, but it'd be nice if the "bake to keyframes" command worked as intended.

**Comment by Gerstmann-Bradley** [comment]
I only tested "bake to keyframe" feature, it seems ok to me to do much beyond frame 250.
Not sure if it's related to your spectrum or node tree, but I feel it should be unlikely?
I am not a developer though, simply curious about your Os, blender version, and whether it's newest version of AN?

**Comment by FlynnD273** [comment]
I am using Blender 2.83, AN version 2.2 on Windows. Is there a way to upload a blend file? If not I can provide it via Google Drive.

**Comment by Gerstmann-Bradley** [comment]
> I am using Blender 2.83, AN version 2.2 on Windows. Is there a way to upload a blend file? If not I can provide it via Google Drive.

you can zip the blender file, and then you will be able to upload it.

**Comment by FlynnD273** [comment]
Turns out my file was over 10MB. Oh well. Here's the Google Drive link: https://drive.google.com/file/d/1VHzugkCaQnQrjpMsvXyFf2uxlJTXI7E-/view?usp=sharing

**Comment by Gerstmann-Bradley** [comment]
> Turns out my file was over 10MB. Oh well. Here's the Google Drive link: https://drive.google.com/file/d/1VHzugkCaQnQrjpMsvXyFf2uxlJTXI7E-/view?usp=sharing

The file size is actually >400 MB, what happened to your file. That's so abnormal.

**Comment by FlynnD273** [comment]
Hmm... I have a wav file, but that could only account for about 40MB. I really don't know. I am using the instancer node to create the spectrum objects. Could it be possible that I'm creating a new set of objects every frame?

**Comment by Gerstmann-Bradley** [comment]
> Hmm... I have a wav file, but that could only account for about 40MB. I really don't know. I am using the instancer node to create the spectrum objects. Could it be possible that I'm creating a new set of objects every frame?

According to your tree, it's unlikely. The amount of instances is set 128. The spectrum only changes its transform.

**Comment by FlynnD273** [comment]
Okay. In that case, I have absolutely no idea where the extra file size comes from. That may be an issue in and of itself, unrelated to AN.

**Comment by Gerstmann-Bradley** [comment]
![conflict](https://user-images.githubusercontent.com/55994236/90345553-86cd4a00-dfef-11ea-94f5-9c611ebc8494.png)

For this part of node tree, two info are conflicting each other. Why are you using matrix output followed by a transform output?

**Comment by FlynnD273** [comment]
Um... I use the matrix output to space out the cubes, then transform them. Is there a better way to do this?

**Comment by Gerstmann-Bradley** [comment]
> Um... I use the matrix output to space out the cubes, then transform them. Is there a better way to do this?

![solution](https://user-images.githubusercontent.com/55994236/90345651-7c5f8000-dff0-11ea-8f1f-5cd94771c8e4.png)

Solution provided. With this solution I also have no problem of baking.
It slows down the execution, but I think it's expected, and your original tree was broken due to the conflict.

Object matrix output and object transform output points to related data, I think you can't connect them together for the same object.

**Comment by Gerstmann-Bradley** [comment]
[FrequencyRangeAnimationNodes debug.zip](https://github.com/JacquesLucke/animation_nodes/files/5081495/FrequencyRangeAnimationNodes.debug.zip)

Please test this file, to see if you have issue of baking

**Comment by FlynnD273** [comment]
Oh wow. That is a lot faster. Thank you so much for the help! It still seems to stop baking the keyframes at frame 250, though... Is there a setting somewhere that I'm completely overlooking?

**Comment by Gerstmann-Bradley** [comment]
> Oh wow. That is a lot faster. Thank you so much for the help! It still seems to stop baking the keyframes at frame 250, though... Is there a setting somewhere that I'm completely overlooking?

Then I have no idea, because your issue is not replicable on my computer with that file.
win10, 2.83.3, AN 2.2
You have to wait Omar for answer.

**Comment by FlynnD273** [comment]
Hm... okay. Thank you for your time. This was really helpful, even if my original problem couldn't be solved.

**Comment by Gerstmann-Bradley** [comment]
A side note, if you are only effect locations. Instead of baking keyframe to all channels (scale, rotation_euler that you are not using), you can use "set keyframes" node and play the animation. More info can be found in mannual.

![side note](https://user-images.githubusercontent.com/55994236/90346207-2fca7380-dff5-11ea-96cd-72d4a94a3ff7.png)

**Comment by FlynnD273** [comment]
Interesting... so the bake to keyframes command will bake everything, not just the properties that change?

**Comment by Gerstmann-Bradley** [comment]
> Interesting... so the bake to keyframes command will bake everything, not just the properties that change?

yes, and you can obviously test it by yourself to know the result. I mention this so it may help reduce your file size. As for whether it solves your unique problem, you also have to test by yourself.

**Comment by OmarEmaraDev** [comment]
I can't replicate the issue either. Notice that this is a modal operator that can be ended by pressing certain keys like Esc and Right Click. Might there be another add-on that conflicts with Animation Nodes? Can you try turning off other add-ons and see if it works?

**Comment by FlynnD273** [comment]
I just tried turning off various addons, with no luck. The only addon that I had enabled that wasn't enabled from installation (cycles render, STL export, etc.) was the fSpy camera data importer, and I disabled that one too. Even on an entirely new, blank file, it won't cache automatically past frame 250. The only thing I did to the file is change the end frame from 250 to 425.

**Comment by Gerstmann-Bradley** [comment]
> I just tried turning off various addons, with no luck. The only addon that I had enabled that wasn't enabled from installation (cycles render, STL export, etc.) was the fSpy camera data importer, and I disabled that one too. Even on an entirely new, blank file, it won't cache automatically past frame 250. The only thing I did to the file is change the end frame from 250 to 425.

Have you tried to load the factory setting of blender? enable AN and then give it another try?

**Comment by OmarEmaraDev** [comment]
Do you have multiple scenes? Can you apply this diff and show us the terminal output after executing the bake?

```diff
diff --git a/animation_nodes/operators/bake.py b/animation_nodes/operators/bake.py
index b4be54e8..bedc48f5 100644
--- a/animation_nodes/operators/bake.py
+++ b/animation_nodes/operators/bake.py
@@ -11,6 +11,7 @@ class BakeAnimation(bpy.types.Operator):
     endFrame: IntProperty(default = 250)

     def invoke(self, context, event):
+        print(self.endFrame)
         context.window_manager.modal_handler_add(self)
         self.timer = context.window_manager.event_timer_add(0.001, window = context.window)

```

**Comment by FlynnD273** [comment]
Huh. As soon as I added that line, it works perfectly. I have no idea what happened. It may be that reloading my preferences reset something that made it not work? I'm not sure. At any rate, it now works fine. Though I haven't been able to find the output for the bake function. Should I be looking at the python console? Or something else?

**Comment by OmarEmaraDev** [comment]
Yes, the Python console. But now that it works, that's probably not very useful now.

**Comment by FlynnD273** [comment]
Well, I still don't understand why I'm not seeing the console output... The diff file you posted only added the print statement, right?

**Comment by OmarEmaraDev** [comment]
Yes.
